src/VisualStudio/Core/Impl/SolutionExplorer/AnalyzerItem/AnalyzerItemSource.cs

```diff
@@ -2,237 +2,169 @@
-        private BulkObservableCollection<AnalyzerItem> _analyzerItems;
-
-        public event PropertyChangedEventHandler PropertyChanged;
-
-        public AnalyzerItemSource(AnalyzersFolderItem analyzersFolder, IAnalyzersCommandHandler commandHandler)
-        {
-            _analyzersFolder = analyzersFolder;
-            _commandHandler = commandHandler;
-
-            _analyzersFolder.Workspace.WorkspaceChanged += Workspace_WorkspaceChanged;
-        }
-
-        private void Workspace_WorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
-        {
-            switch (e.Kind)
-            {
-                case WorkspaceChangeKind.SolutionAdded:
-                case WorkspaceChangeKind.SolutionChanged:
-                case WorkspaceChangeKind.SolutionReloaded:
-                    UpdateAnalyzers();
-                    break;
-
-                case WorkspaceChangeKind.SolutionRemoved:
-                case WorkspaceChangeKind.SolutionCleared:
-                    _analyzersFolder.Workspace.WorkspaceChanged -= Workspace_WorkspaceChanged;
-                    break;
-
-                case WorkspaceChangeKind.ProjectAdded:
-                case WorkspaceChangeKind.ProjectReloaded:
-                case WorkspaceChangeKind.ProjectChanged:
-                    if (e.ProjectId == _analyzersFolder.ProjectId)
-                    {
-                        UpdateAnalyzers();
-                    }
-
-                    break;
-
-                case WorkspaceChangeKind.ProjectRemoved:
-                    if (e.ProjectId == _analyzersFolder.ProjectId)
-                    {
-                        _analyzersFolder.Workspace.WorkspaceChanged -= Workspace_WorkspaceChanged;
-                    }
-
-                    break;
-            }
-        }
-
-        private void UpdateAnalyzers()
-        {
-            if (_analyzerItems == null)
-            {
-                // The set of AnalyzerItems hasn't been realized yet. Just signal that HasItems
-                // may have changed.
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
 
-                NotifyPropertyChanged(nameof(HasItems));
-                return;
-            }
+internal sealed class AnalyzerItemSource : IAttachedCollectionSource
+{
+    private readonly AnalyzersFolderItem _analyzersFolder;
+    private readonly IAnalyzersCommandHandler _commandHandler;
 
-            var project = _analyzersFolder.Workspace
-                            .CurrentSolution
-                            .GetProject(_analyzersFolder.ProjectId);
+    private readonly BulkObservableCollection<AnalyzerItem> _items = [];
 
-            if (project != null &&
-                project.AnalyzerReferences != _analyzerReferences)
-            {
-                _analyzerReferences = project.AnalyzerReferences;
+    private readonly CancellationTokenSource _cancellationTokenSource = new();
+    private readonly AsyncBatchingWorkQueue _workQueue;
```

Comment:
switched to a work queue model.  We listen to workspace vents, which then just pulse the queue.